### ISSUE TITLE
Customize the Niagara button 

### ISSUE THREAD

**Issue by iankyburz** [8bitPit/Niagara-Issues#2672]
### Is your feature request related to a problem?

My Niagara button shows kind of a ereader icon for my e-mail app. I would like to switch to a e-mail icon.

### Describe the solution you'd like

Just a simple function to chose another icon

### Feature Request Procedure

- [X] I have read the [knowledge base article on feature requests](https://help.niagaralauncher.app/article/44-feature-requests) and followed the steps mentioned

**Comment by iankyburz** [comment]
I'm adding a screenshot of my Niagara button and another screenshot of the e-mail app that is attached to the button.
![Screenshot_20240603-115350](https://github.com/8bitPit/Niagara-Issues/assets/168526607/dc5326e6-3a45-4940-944b-7b94d61ab90a)
![Screenshot_20240603-115455](https://github.com/8bitPit/Niagara-Issues/assets/168526607/ea607023-bea4-472a-8a6a-d3a5b00a12f5)

**Comment by iankyburz** [comment]
To make it clear: eM Client is my e-mail app and definitely not an ebook reader.

**Comment by contact-maximux** [comment]
long tap on icon, then click the picture of icon, you can change from there

**Comment by iankyburz** [comment]
This doesn't work for the Niagara button. There I can only change the underlaying app but not the icon.

**Comment by iankyburz** [comment]
This is how the Niagara button looks like when I'm using Gmail as underlying app. With "eM Client" it should look the same. 
![Screenshot_20240603-192438](https://github.com/8bitPit/Niagara-Issues/assets/168526607/76de7c0a-06dd-42fb-9730-e0e18d1539ff)

**Comment by contact-maximux** [comment]
SO that means Niagra button not following Icon pack theme

**Comment by iankyburz** [comment]
No. It's not about the theme, it's about the app category. Niagara choose the wrong category for my e-mail app.

**Comment by TFackler** [comment]
This sounds like a good idea, but we would prefer if the categorization algorithm did this for our users instead of them having to select the icon manually.

We will think about whether this is a viable option or whether we stick to the original idea of improving the icon selection algorithm.

**Comment by rudism** [comment]
It seems like a simple strategy would be to use the same icon as whatever app is set as the tap action (either use the icon for that app's category, or if the icon of that app has been manually adjusted to something else then use that icon).

**Comment by TFackler** [comment]
From #2757 - @camina-drummer
> [...] I'd prefer manual selection.

**Comment by TFackler** [comment]
@errantepiphany 

That would be close to what we wanted to add.

The only issues are that the categorization algorithm requires the user to grant the app categorization permission and that not every icon would fit on the Niagara button since it's a floating action button. We could allow users to choose any of the Anycons, both individual and category icons.

**Comment by errantepiphany** [comment]
@TFackler I thought that app icons in Android are standardized/uniform and even scalable, so it shouldn't matter if they are the original app icons, Niagara-provided icons, or a custom icon pack. If that's the case, why wouldn't any icon fit on the floating action button? Alternatively, couldn't the floating action button itself scale to match the icon size? Niagara already supports resizing the app icons on the home screen... Can scaling not also be done on the floating action button?

**Comment by Sorrieah** [comment]
Chiming in to say that I'm having an issue with this as well. I use the anycons and have Obsidian as the tap action for my Niagara button, but it shows an open book instead of the Obsidian anycon that I use. I'd really like it to show the anycon for the app/action I've selected.

**Comment by bradtse** [comment]
I have the Google Home app as the click action and it shows me a broken URL icon. Seems like a pretty major/common app to not have support for. An "ugly" icon is much preferred to a useless one. 

![SmartSelect_20250429_164322_Niagara Launcher.jpg](https://github.com/user-attachments/assets/d4bf629b-7373-4412-b53f-07f690adebbf)
